### PR TITLE
[BPK-964] eslint upgrades

### PR DESCRIPTION
**Pull request by shaundon**
Have upgraded eslint and airbnb etc to latest versions.

Have added some opt-outs in `.eslintrc`. Each of them is commented to explain why exactly we're opting out.

Off the back of this, we will upgrade `eslint-config-skyscanner` and `backpack-react-scripts`.

### CODE REVIEW

File: packages/bpk-component-tooltip/stories.js [hunk 1]
```diff
@@ -91,7 +93,9 @@ storiesOf('bpk-component-tooltip', module)
         id="my-tooltip"
         target={<a href="//skyscanner.net" target="_blank" rel="noopener noreferrer">Click me!</a>}
       >
-        Find cheap flights here! ✈️
+        Find cheap flights here!
+        <span role="img" aria-label="Plane">✈️</span>
+
```
Comment (k0nserv) [inline packages/bpk-component-tooltip/stories.js]: Should we remoe this empty line.

File: packages/bpk-component-tooltip/stories.js [hunk 2]
```diff
@@ -91,7 +93,9 @@ storiesOf('bpk-component-tooltip', module)
         id="my-tooltip"
         target={<a href="//skyscanner.net" target="_blank" rel="noopener noreferrer">Click me!</a>}
       >
-        Find cheap flights here! ✈️
+        Find cheap flights here!
+        <span role="img" aria-label="Plane">✈️</span>
```
Comment (k0nserv) [inline packages/bpk-component-tooltip/stories.js]: Was this change caught by the linter?

File: .eslintrc [hunk 3]
```diff
@@ -8,6 +8,37 @@
...
+    "jsx-a11y/label-has-for": [ 2, {
+      "components": [ "Label" ],
+      "required": {
+          "some": [ "nesting", "id" ]
+      },
+      "allowChildren": false,
+    }],
+
+    // To be addressed.
+    "no-buffer-constructor": 0, // Need to migrate to new style.
```
[the remark this answers is not in the document]
Comment (shaundon) [reply .eslintrc]: It's used quite heavily in the SVG building from what I can see

File: packages/bpk-component-image/src/BpkImage-test.js [hunk 4]
```diff
@@ -89,10 +89,10 @@ describe('BpkImage', () => {
         width={816}
         height={544}
         src="./path/to/image_1640.jpg"
-        srcSet={`./path/to/image_320px.jpg 320w,
+        srcSet="./path/to/image_320px.jpg 320w,
```
[the remark this answers is not in the document]
Comment (shaundon) [reply packages/bpk-component-image/src/BpkImage-test.js]: I think the autofix got this, we can change it back if we prefer.

File: packages/bpk-component-loading-button/src/BpkLoadingButton.js [hunk 5]
```diff
@@ -136,6 +136,7 @@ BpkLoadingButton.defaultProps = {
   secondary: false,
   selected: false,
   destructive: false,
+  link: false,
```
[the remark this answers is not in the document]
Comment (shaundon) [reply packages/bpk-component-loading-button/src/BpkLoadingButton.js]: I don't think it changes behaviour, the only way it's used is:

```if (props.link) { classNames.push(getClassName('bpk-loading-button--link')); }```

File: packages/bpk-component-tooltip/stories.js (hunk 2 above)
Comment (shaundon) [reply packages/bpk-component-tooltip/stories.js]: It was! See here: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/accessible-emoji.md

File: .eslintrc (hunk 3 above)
[the remark this answers is not in the document]
Comment (k0nserv) [reply .eslintrc]: Is that our code?

File: packages/bpk-component-image/src/BpkImage-test.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (k0nserv) [reply packages/bpk-component-image/src/BpkImage-test.js]: We could put it in a variable created with a template literal, but meh it's good enough

File: packages/bpk-component-tooltip/stories.js (hunk 2 above)
Comment (k0nserv) [reply packages/bpk-component-tooltip/stories.js]: That's great :)

File: packages/bpk-component-calendar/src/BpkCalendarNav.js [hunk 6]
```diff
@@ -72,7 +74,9 @@ const BpkCalendarNav = (props) => {
           type="button"
           className={getClassName('bpk-calendar-nav__button')}
           id={`${id}_month_nudger_previous`}
-          onClick={changeMonth({ month: prevMonth, min, max, callback: onMonthChange, source: 'PREV' })}
+          onClick={changeMonth({
+ month: prevMonth, min, max, callback: onMonthChange, source: 'PREV',
```
[the remark this answers is not in the document]
Comment (shaundon) [reply packages/bpk-component-calendar/src/BpkCalendarNav.js]: Yeah that does look weird! I'll take another look at that one.

File: .eslintrc [hunk 7]
```diff
@@ -8,6 +8,37 @@
     "max-len": [2, 120],
     "react/jsx-filename-extension": 0,
     "import/no-extraneous-dependencies": 0,
-    "react/prefer-stateless-function": [2, { "ignorePureComponents": true }]
+    "react/prefer-stateless-function": [2, { "ignorePureComponents": true }],
+
+    // Addressed.
+
+    // We've decided we don't want this for now as it makes writing JSX more painful.
+    "function-paren-newline": 0,
```
Comment (k0nserv) [inline .eslintrc]: What value does the config have for this? I don't think fully disabling it is a good idea. The `minItems` options seems like what we want. Any examples of how it makes JSX painful?

File: packages/bpk-component-calendar/src/BpkCalendarNav.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (shaundon) [reply packages/bpk-component-calendar/src/BpkCalendarNav.js]: Fixed

File: .eslintrc [hunk 8]
```diff
@@ -8,6 +8,29 @@
     "max-len": [2, 120],
     "react/jsx-filename-extension": 0,
     "import/no-extraneous-dependencies": 0,
-    "react/prefer-stateless-function": [2, { "ignorePureComponents": true }]
+    "react/prefer-stateless-function": [2, { "ignorePureComponents": true }],
+
+    // We've decided we don't want this for now as it makes writing JSX more painful.
+    "function-paren-newline": 0,
```
Comment (matthewdavidson) [inline .eslintrc]: Can we put a link to the issue here:

https://github.com/airbnb/javascript/issues/1584#issuecomment-335667272

File: packages/bpk-component-image/src/BpkImage-test.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (shaundon) [reply packages/bpk-component-image/src/BpkImage-test.js]: I'm changing it to a variable :)

File: .eslintrc (hunk 8 above)
Comment (shaundon) [reply .eslintrc]: Done

**Pull request merged**